src/utils/extractVideoEmbedFromStory.ts

```diff
@@ -0,0 +1,16 @@
+const extractVideoEmbedFromStory = (content) => {
+  if (!content || !content.type) return undefined;
+  const contentEl = (
+    (content.type === 'video' && content) || (
+      content.type === 'story'
+      && content.promo_items
+      && content.promo_items.lead_art
+      && content.promo_items.lead_art.type
+      && content.promo_items.lead_art.type === 'video'
+        ? content.promo_items.lead_art : undefined
+    )
+  );
+  return contentEl ? contentEl.embed_html : undefined;
```

Comment:
Had to rewrite this to understand what was going on.

```ts
const extractVideoEmbedFromStory = (content): string => {
  if (content && content.type === 'video') {
    return content.embed_html;
  }

  if (
    (content && content.type === 'story')
    && (
      content.promo_items
      && content.promo_items.lead_art
      && content.promo_items.lead_art.type === 'video'
    )
  ) {
    return content.promo_items.lead_art.embed_html;
  }

  return undefined;
};
```

Does this track with what you were going for? Tests pass for this version so I assume so. Damn shame optional chaining isn't working with our Jest setup.

Reply:
Yeah I agree - I actually had this written with optional chaining before and then rewrote it without. I think the way you have it re-written replicates the logic but I will test it out locally to make sure everything still works as expected on the front-end 👍